binding.cc

```diff
@@ -1043,6 +1096,92 @@ NAPI_METHOD(db_del) {
   NAPI_RETURN_UNDEFINED();
 }
 
+/**
+ * Worker class for deleting a range from a database.
+ */
+struct ClearWorker final : public PriorityWorker {
+  ClearWorker (napi_env env,
+               Database* database,
+               napi_value callback,
+               bool reverse,
+               int limit,
+               std::string* lt,
+               std::string* lte,
+               std::string* gt,
+               std::string* gte)
+    : PriorityWorker(env, database, callback, "leveldown.db.clear"),
+      limit_(limit),
+      count_(0) {
+    baseIterator_ = new BaseIterator(database, reverse, lt, lte, gt, gte, false);
+    writeOptions_ = new leveldb::WriteOptions();
+    writeOptions_->sync = false;
+  }
+
+  ~ClearWorker () {
+    // TODO: write GC tests
+    delete baseIterator_;
+    delete writeOptions_;
+  }
+
+  void DoExecute () override {
+    baseIterator_->Initialize();
+
+    while (baseIterator_->Valid()) {
+      leveldb::Slice keySlice = baseIterator_->CurrentKey();
+
+      if ((limit_ >= 0 && ++count_ > limit_) || baseIterator_->OutOfRange(keySlice)) {
+        break;
+      }
+
+      // TODO: use batches
+      leveldb::Status delStatus = database_->Del(*writeOptions_, keySlice);
```

Comment:
Doing the delete natively (without crossing the JS/C++ boundary) should already be much faster, but I didn't benchmark it yet because I first want to introduce batching here.